journals/apps/journals/management/commands/gather_videos.py

```diff
@@ -16,15 +16,30 @@
 from django.core.management.base import BaseCommand
 from wagtail.wagtailcore.models import Collection, Site
 
-from journals.apps.journals.models import Journal, Video
+from journals.apps.journals.models import Journal, Video, JournalPage
 
 
 class Command(BaseCommand):
-    '''Management command to gather course videos'''
+    """
+    Management command to gather course videos
+    """
     help = 'Gathers all videos from relevant courses'
 
-    def rewrite_url_for_external_use(self, url, site):
-        '''Updates domain of URLs to use external host in declared in settings'''
+    def rewrite_url_for_external_use(self, url, site, block_id):
+        """
+        Updates domain of URLs to use external host in declared in settings
+        """
+        journal_pages = JournalPage.objects.filter(videos=Video.objects.get(block_id=block_id))
```

Comment:
this is not going to work as the video is not associated with any journal_pages when it is imported. it is only assoicated with a page after it is actually added to the page but that doesn't happen in this mgmt command. Leave this as it was before and make a change in blocks.py XBlockVideoBlock.get_api_repsentation() to return view_url based on the logic here.

Reply:
@bill-filler but I have to get the journal id from the video object. what is the best way to get that ?

Reply:
@hasnain-naveed 
I tested and it does not work doing this in gather videos. You need to move the logic to the block